### DISCUSSION TITLE
How can I show a Music Note before the song titles?

### DISCUSSION THREAD

**Discussion by ajtruckle** [ajtruckle/meeting-schedule-assistant-support#27] [Q&A] [answered]
Can we include a music note symbol in our meeting schedules? Similar to the meeting workbooks?

<img width="554" alt="Screenshot 2023-11-29 at 05 11 19" src="https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/69472f12-bbd5-4064-8697-f1cb6002b046">

**Comment by ajtruckle** [comment, marked as answer]
There are a couple of ways that this can be done:

- Image
- Font Awesome

<hr>

### Image
This is the easiest to implement IMHO.

1. Download this [music note](https://icon-icons.com/icon/note-audio-music/2535) icon and save it to the `%appdata%\meeting schedule assistant` folder.
2. Open your customised XSL template and update this bit:

```xsl
<!--Displays the song information-->
<xsl:template match="SongOpen | SongMiddle | SongEnd">
  <td class="cellTime">
    <xsl:call-template name="DisplayTime"/>
  </td>
  <td class="cellTheme">
    <xsl:if test="self::SongMiddle">
      <xsl:attribute name="colspan">3</xsl:attribute>
      <xsl:attribute name="class"></xsl:attribute>
    </xsl:if>
    <img src="musicnote.png" width="12"/>
    <xsl:if test="self::SongOpen or self::SongMiddle or (self::SongEnd and . != '0')">
      <span class="textSongLabel">
        <xsl:value-of select="//Labels/Song"/>
      </span>
      <span class="textSongNumber">
        <xsl:value-of select="."/>
      </span>
    </xsl:if>
    <span class="textSongTitle">
      <xsl:value-of select="@Title"/>
    </span>
  </td>
  <xsl:if test="self::SongOpen or self::SongEnd">
    <td class="cellPosition">
      <xsl:value-of select="//Labels/Prayer"/>
    </td>
  </xsl:if>
</xsl:template>
```

What I have done is remove the **bullet** and replace it with the music note image:

`<img src="musicnote.png" width="12"/>`

I resize it to 12 pixels since the original is 32 pixels. A better approach is to resize the image itself so it is the right size.

Sample:

![sample music note with image](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/afb9f4d9-1ee1-420c-a242-35ad9a5e178c)

<hr>

### Font Awesome

A couple of steps are involved to use the free version of the Font Awesome [music note](https://fontawesome.com/icons/music?f=classic&s=solid):

1. Add a link to the Font Awesome CSS file:

`<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/all.min.css">`

2. Wherever you want a music symbol you would use:

`<i class="fa-solid fa-music"></i>`

Sample:

![sample music note with font awesome](https://github.com/ajtruckle/meeting-schedule-assistant-support/assets/7985331/48481583-c35e-4200-a89c-70216d150940)

> [!NOTE]
Several steps are required for using the _Font Awesome_ approach.

- Step 1
Add a `CSSFile2` parameter towards the top, like this:

```xsl
<?xml version="1.0" encoding="utf-8"?>
<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns="http://www.w3.org/1999/xhtml">
	<xsl:output method="html" indent="yes" version="4.01"
	  doctype-system="http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"
	  doctype-public="-//W3C//DTD XHTML 1.0 Transitional//EN"/>
	<xsl:param name="CSSFile1"></xsl:param>
	<xsl:param name="CSSFile2"></xsl:param>
	<xsl:template match="/">

```

- Step 2
Adjust the `<head>` section as follows:

```xsl
<head>
	<meta http-equiv="X-UA-Compatible" content="IE=edge" />
	<meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
	<xsl:choose>
		<xsl:when test="$CSSFile1 !='' and $CSSFile2 !=''">
			<style type="text/css">
				<xsl:value-of select="$CSSFile1"/>
				<xsl:value-of select="$CSSFile2"/>
			</style>
		</xsl:when>
		<xsl:otherwise>
			<link rel="stylesheet" type="text/css" href="Workbook-S-140.css"/>
			<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.2/css/all.min.css"/>
		</xsl:otherwise>
	</xsl:choose>
	<title>
		<xsl:value-of select="//Labels/Congregation"/>&#160;<xsl:value-of select="//Labels/Title" />
	</title>
</head>
```

- Step 3
Use `<i class="fa-solid fa-music"></i>` as previously instructed to show the music note symbol. But you might have to fine-tune some of the other classes if the font gets affected.

**Reply by MardoDan** [reply]
Excelente trabajo @ajtruckle Va quedando igual que el Modelo S-140 y la Guía de Actividades de JW.
